net/xia/ppal_ether/main.c

```diff
@@ -297,13 +361,18 @@ static int main_newroute(struct xip_ppal_ctx *ctx, struct fib_xid_table *xtbl,
 	fxid_init(xtbl, &mether->xem_common, id, XRTABLE_MAIN_INDEX, 0);
 	xdst_init_anchor(&mether->xem_anchor);
 	INIT_LIST_HEAD(&mether->interface_common_addr);
+	seqlock_init(&mether->cached_hdr.hh_lock);
```

Comment:
This line is better placed in `xia_ether_header_cache()`, where we are initializing the whole `struct hh_cache`.